src/index.d.ts

```diff
@@ -9,8 +9,11 @@ interface SvelteTableColumn {
   component?: typeof SvelteComponent;
   helpModal?: typeof SvelteComponent;
   value?: (v: unknown) => string;
+  titleComponent: typeof SvelteComponent;
```

Comment:
Should title and titleComponent be optional?

Reply:
I added the rule that one of them (title or titleComponent is optional), check below. 
If we specify as title?:..., titleComponent?:... that will mean that the user can use the table without both of them.